File: client/src/types/SeriesMapping.ts
import type { components } from "@/lib/api-types";
import type { Require } from "@/lib/dto";

// AudiobookManager.Domain/SeriesMapping.cs: regex/warnAboutPart are non-nullable; id is nullable —
// the same model doubles as the create-request body (id: null) and the list/update-response body
// (id always populated for a persisted row), which is why this file splits it into a request-shaped
// base and a response-shaped extension instead of aliasing the schema directly. There is no target
// field: a pattern belongs to a Series (the owning page's series), whose name is its target.
export type SeriesMappingBase = Require<
  components["schemas"]["SeriesMapping"],
  "regex" | "warnAboutPart"
>;

export interface SeriesMapping extends SeriesMappingBase {
  id: number;
}

File: AudiobookManager/AudiobookManager.Domain/SeriesMapping.cs
namespace AudiobookManager.Domain;

/// <summary>
/// One regex mapping pattern owned by a series. The pattern has no target of its own: when the
/// incoming metadata series value matches <see cref="Regex"/>, <see cref="AudiobookManager.Scraping.BookSeriesMapper"/>
/// rewrites it to the owning series' name. <see cref="Id"/> is null only while the mapping has not
/// been persisted (the same model doubles as the create-request body).
/// </summary>
public class SeriesMapping
{
    public long? Id { get; set; }
    public string Regex { get; set; }
    public bool WarnAboutPart { get; set; }

    public SeriesMapping(long? id, string regex, bool warnAboutPart)
    {
        Id = id;
        Regex = regex;
        WarnAboutPart = warnAboutPart;
    }
}
